addons/l10n_in/models/account_invoice.py

```diff
@@ -13,21 +14,60 @@ def _compute_amount_total_words(self):
+    l10n_in_gstin = fields.Char(string="GSTIN", readonly=True, states={'draft': [('readonly', False)]})
     # For Export invoice this data is need in GSTR report
-    l10n_in_export_type = fields.Selection([
-        ('regular', 'Regular'), ('deemed', 'Deemed'),
-        ('sale_from_bonded_wh', 'Sale from Bonded WH'),
-        ('export_with_igst', 'Export with IGST'),
-        ('sez_with_igst', 'SEZ with IGST payment'),
-        ('sez_without_igst', 'SEZ without IGST payment')],
-        string='Export Type', default='regular', required=True)
     l10n_in_shipping_bill_number = fields.Char('Shipping bill number', readonly=True, states={'draft': [('readonly', False)]})
     l10n_in_shipping_bill_date = fields.Date('Shipping bill date', readonly=True, states={'draft': [('readonly', False)]})
-    l10n_in_shipping_port_code_id = fields.Many2one('l10n_in.port.code', 'Shipping port code', states={'draft': [('readonly', False)]})
+    l10n_in_shipping_port_code_id = fields.Many2one('l10n_in.port.code', 'Port code', states={'draft': [('readonly', False)]})
     l10n_in_reseller_partner_id = fields.Many2one('res.partner', 'Reseller', domain=[('vat', '!=', False)], help="Only Registered Reseller", readonly=True, states={'draft': [('readonly', False)]})
-    l10n_in_partner_vat = fields.Char(related="partner_id.vat", readonly=True)
+
+    @api.constrains('l10n_in_gstin', 'journal_id')
+    def _check_l10n_in_gstin(self):
+        check_vat_in = self.env['res.partner'].check_vat_in
+        wrong_gstin_moves = self.filtered(lambda move:
+            move.l10n_in_company_country_code == 'IN' and
+            move.l10n_in_gst_treatment in ['regular', 'composition', 'special_economic_zone', 'deemed_export'] and
+            move.l10n_in_gstin != False and
+            not check_vat_in(move.l10n_in_gstin))
+        if wrong_gstin_moves:
+            raise ValidationError(_("The GSTIN number [%s] does not seem to be valid")%(",".join(m.l10n_in_gstin for m in wrong_gstin_moves)))
+
+    @api.onchange('partner_id')
+    def _onchange_partner_id(self):
+        """Use journal type to define document type because not miss state in any entry including POS entry"""
+        if self.l10n_in_company_country_code == 'IN':
+            self.l10n_in_gst_treatment = self.partner_id.l10n_in_gst_treatment
+            self._onchange_l10n_in_gst_treatment()
+        return super()._onchange_partner_id()
+
+    @api.onchange('l10n_in_gst_treatment')
+    def _onchange_l10n_in_gst_treatment(self):
+        if self.l10n_in_company_country_code == 'IN':
+            if self.l10n_in_gst_treatment == self.partner_id.l10n_in_gst_treatment:
+                self.l10n_in_gstin = self.partner_id.vat
+            else:
+                self.l10n_in_gstin = False
+
+    @api.model
+    def _check_l10n_in_state(self, partner):
+        """If sate is not India's state then return Other Territory with code 97"""
```

Comment:
Yes we have, because government rules say that set place of supply to "Other Territory" with code 97 when export so it is there.
you can check it here https://einv-apisandbox.nic.in/state-code.html

Reply:
But that can be done in the API.  Because it is possible that the address of the partner will have a state in its own country.

Reply:
Partner can set own country state.
here we have specific state_id in invoice so we set that value in invoice so no problem.

Reply:
Improve comment...  (also explaining why this other territory state is needed) + check is not logical as method name --> it returns, so get or something would be more logical.